File: scripts/fix_spacer.dart
import 'dart:io';

void main() {
  final dir = Directory('lib/features/kids_zone/presentation/pages/games');
  final files = dir
      .listSync(recursive: true)
      .whereType<File>()
      .where((f) => f.path.endsWith('.dart'))
      .toList();

  // also check kids_zone_screen.dart and buddy_boutique_screen.dart, sticker_book_screen.dart
  files.addAll([
    File('lib/features/kids_zone/presentation/pages/kids_zone_screen.dart'),
    File(
      'lib/features/kids_zone/presentation/pages/buddy_boutique_screen.dart',
    ),
    File('lib/features/kids_zone/presentation/pages/sticker_book_screen.dart'),
  ]);

  for (final file in files) {
    if (!file.existsSync()) continue;
    String content = file.readAsStringSync();

    if (content.contains('const Spacer()')) {
      // Find Padding( padding: EdgeInsets.symmetric(horizontal: 24.w), child: Column(...
      // We want to turn the Column into a constrained container

      bool modified = false;

      // Replace Spacer
      if (content.contains('const Spacer(),')) {
        content = content.replaceAll(
          'const Spacer(),',
          'SizedBox(height: 40.h),',
        );
        modified = true;
      }

      // If it's a typical game screen with Padding -> child: Column
      if (content.contains('child: Column(\n                children: [')) {
        content = content.replaceAll(
          'child: Column(\n                children: [',
          'child: Container(\n                constraints: BoxConstraints(\n                  minHeight: MediaQuery.of(context).size.height - 200.h,\n                ),\n                child: Column(\n                  mainAxisAlignment: MainAxisAlignment.spaceBetween,\n                  children: [',
        );

        // Fix the closing tags. Wait, it's safer to just wrap the Column if we know the exact indentation.
        // Let's just do a simple replacement for the opening of the Column, and then find the corresponding closing.
        // Actually, wait, replacing `child: Column(` to `child: Container(\n constraints..., child: Column(` means we need to add another `),` at the end of the Column.
        // This is a bit risky with pure string replacement if the indentation varies.
      }

      if (modified) {
        // Instead of structural replacement, maybe just replacing Spacer with a large SizedBox or Expanded and adjusting list view is better?
        // Wait, the crash is because Column shrinkWraps inside a SingleChildScrollView, meaning we cannot use Expanded or Spacer. Spacer is essentially Expanded(child: SizedBox.shrink()).
        // If we just replace const Spacer() with SizedBox(height: 40.h), the layout won't crash and might look okay! Let's check.
      }
    }
  }
}

File: scripts/fix_spacer.py
import os
import glob

# Path to the kids zone games directory
base_dir = r"c:\Users\asus\Documents\App Projects\voxai_quest\lib\features\kids_zone\presentation\pages"

def fix_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    if "const Spacer()" not in content:
        return

    # Replace Spacer
    content = content.replace("const Spacer(),", "SizedBox(height: 40.h),")

    # Wrap Column with Container and BoxConstraints
    # We find the specific pattern
    target_pattern = """            child: Column(
              children: ["""
              
    replacement = """            child: Container(
              constraints: BoxConstraints(
                minHeight: MediaQuery.of(context).size.height - 200.h,
              ),
              child: Column(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: ["""
                
    if target_pattern in content:
        content = content.replace(target_pattern, replacement)
        
        # Now we need to add a closing bracket '),' for the Container.
        # It's usually closed like this:
        #             ),
        #           ),
        #         ],
        #       );
        # We can just look for the end of the Padding which wraps the Column.
        # But a simpler way: find the exact closing pattern of the Stack which encapsulates it.
        # In numbers_game_screen it was:
        #                 ],
        #               ),
        #             ),
        #           ],
        #         );
        # Let's replace the ending of the Column.
        end_pattern = """                ],
              ),
            ),
          ],
        );"""
        
        end_replacement = """                ],
              ),
            ),
          ),
        ],
      );"""
      
        if end_pattern in content:
            content = content.replace(end_pattern, end_replacement)
            print(f"Fixed structural issue in {filepath}")
        else:
            print(f"End pattern not found in {filepath}. Used fallback.")
            
    # Also check varying indentations
    target_pattern2 = """              child: Column(
                children: ["""
    replacement2 = """              child: Container(
                constraints: BoxConstraints(
                  minHeight: MediaQuery.of(context).size.height - 200.h,
                ),
                child: Column(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: ["""
    end_pattern2 = """                  ],
                ),
              ),
            ],
          );"""
    end_replacement2 = """                  ],
                ),
              ),
            ),
          ],
        );"""
        
    if target_pattern2 in content:
        content = content.replace(target_pattern2, replacement2)
        if end_pattern2 in content:
            content = content.replace(end_pattern2, end_replacement2)
            print(f"Fixed structural issue in {filepath} (Indent 2)")
            
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

# Process all dart files
for filepath in glob.glob(os.path.join(base_dir, "**", "*.dart"), recursive=True):
    fix_file(filepath)
